### PR TITLE
attack_tk on singularities now made more of a developed feature

### LINKED ISSUE #15883: Touching the singularity with TK gibs you
(WEB REPORT BY: aparist REMOTE: 172.93.109.202:7777)
# Revision
4b92c53739e82525ac2443f7158ae1834bcdb225

# Description
Touching the signularity with TK gibs you

# Steps to Reproduce
Get TK
Get a singularity (only tested with a stage 3 one)
# What you Expected
No reaction, since I used my mind to touch it telephatically

# What Actually Happened
I gibbed on the spot

### PR DESCRIPTION
**Pull request by MadmanMartian**
closes #15883 

Now you get some more effects and flavor text when trying to use telekinesis on the naked singularity of a black hole/some weird blood-obsessed demigod. 

Either it acts like you've looked inside the ark of the covenant, or you've lost a staring contest with a scanner telepath, or you're using Pim Particles without a proper suit.

~~May develop on this more in another PR with some spriter assistance~~ Thank you happycatface for the sprite and idea

Losetta Stone - A stone that, once touched/licked, lets you speak in a language foreign to you. Has to be pulled from beyond the event horizon back into this plane of reality through interacting with the singularity via TK

### CODE REVIEW

**Comment by DeityLink** [comment]
the .dust() doesn't make sense for a regular singulo. For Nar-Sie you could trigger the same animation that appears when someone gets sacrified, and gib instead of dust too. (singulos and narsie are bound to get completely separated in the future)

**Comment by MadmanMartian** [comment]
it is a mystery

**Comment by Shadowmech88** [comment]
I think for singulos, it'd make the most sense for it to just suck the player's mind in, ghosting the player while leaving their body unharmed.

**Comment by Knightkiller11** [comment]
I do like the idea of this, given that I sorta suggested it back in the issue thread, though wouldn't these methods completely remove the player from the game? At least standard death allows for cloning and general ressurection.

Also is Nar-Sie just a reskinned singularity with some unique behaviours? thinking about his movement pattern it makes sense but I wouldn't have guessed that myself.

**Comment by SonixApache** [comment]
@Knightkiller11 
Touching the singularity generally has permanent consequences for your health.
As for the narnar part; https://github.com/vgstation-coders/vgstation13/blob/51df013603bef38819234063a90f4a944215c65b/code/game/gamemodes/cult/narsie.dm#L7

**Comment by DeityLink** [comment]
@Knightkiller11 NarSie was in singularity.dm for the longest time until I moved it. Nar-Sie was designed as a size 9 singularity.
Aside from its movement, and the fact that it affects all atoms in a certain range every step, they have nothing in common. Nar-Sie doesn't emit radiations.

**Comment by MadmanMartian** [comment]
Generally hugging a singularity is considered a one-way ticket to oblivion, so it's probably not the best idea.

**Comment by Intigracy** [comment]
`though wouldn't these methods completely remove the player from the game?`

Suggested fix: Don't touch the singularity.

**Comment by Exxion** [comment]
Honestly I don't see why TKing the singularity should kill you
But looks like I'm in the minority here

**Comment by ghost** [comment]
I mean, a player TK touching the singulo probably only happened once, ever, and it probably won't happen again, so I'm not concerned

**Comment by DeityLink** [comment]
I agree with you @Exxion actually

**Comment by Intigracy** [comment]
You could make it throw them towards the singularity.

**Comment by Exxion** [comment]
On the one hand, that's kind of neat, and also adds a reason for it to kill you if you try
On the other hand, I haven't looked at the sprite yet but there's essentially no chance it isn't fucking loss.jpg on a rock

**Comment by Intigracy** [comment]
no idea what you mean :^)

**Comment by MadmanMartian** [comment]
thanks for spoiling it, inti

**Comment by SonixApache** [comment]
stone 👍 
pulling it out of the singularity with telekinesis 😕

**Comment by Aurx** [comment]
We're /vg/station, not /v/station.

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -94,6 +94,29 @@
 /obj/machinery/singularity/Crossed(atom/movable/A)
 	consume(A)
 
+/obj/machinery/singularity/attack_tk(mob/user)
+	to_chat(user, "<span class = 'notice'>You attempt to comprehend \the [src]...</span>")
+	spawn(rand(50,110))
```
Comment (Iamgoofball) [inline code/modules/power/singularity/singularity.dm:99]: why is this spawn

@PJB3005 did you guys port timer subsystem at all

**Comment by Iamgoofball** [comment]
this is pretty funny, i like it and im going to port it

**Comment by MadmanMartian** [comment]
Go right on ahead, seems more fitting with the /tg/station level of weird shenanigans anyway

**Comment by Iamgoofball** [comment]
vg has the same weird shenanigans

keep the pr in it's current state, this is great

File: code/modules/power/singularity/singularity.dm (hunk 1 above)
Comment (Exxion) [reply code/modules/power/singularity/singularity.dm:99]: No we didn't
But why the fuck would we use it for this when `spawn()` does a perfectly good job?

Comment (PJB3005) [reply code/modules/power/singularity/singularity.dm:99]: TG has to dead horse beat every minor performance gain instead of being sane.

Comment (Exxion) [reply code/modules/power/singularity/singularity.dm:99]: I have a hard time believing that any timer system implemented inside BYOND will have better performance than an internal function

Comment (PJB3005) [reply code/modules/power/singularity/singularity.dm:99]: Eh this is BYOND we're talking about to be fair.

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -94,6 +94,29 @@
...
+					user.dust()
+				else
+					user.visible_message("<span class = 'danger'>\The [user] explodes!</span>")
+					..()
+			else
+				to_chat(user, "<span class = 'notice'>...and manage to grab onto something from the depths of \the [src]!</span>")
+				if(do_after(user, src, 30))
+					to_chat(user, "<span class = notice'>You manage to pull something from beyond to within normal space!</span>")
+					var/obj/structure/losetta_stone/L = new
+					L.alpha = 0
```
Comment (PJB3005) [inline code/modules/power/singularity/singularity.dm:125]: > using animate to fade in objects.

👏 never thought I'd see anybody do this after me doing it in clock cult. This is the kind of quality we need more.

File: code/modules/research/xenoarchaeology/artifact/artifact_losetta.dm [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/obj/structure/losetta_stone
+	name = "strange stone"
+	desc = "You feel a strange sense of emptiness, as if missing something, from looking upon this stone."
+	icon = 'icons/obj/xenoarchaeology.dmi'
+	icon_state = "loss"
+	var/chosen_language
+
+/obj/structure/losetta_stone/New()
+	..()
+	chosen_language = pick(LANGUAGE_HUMAN,\
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/artifact_losetta.dm]: I believe you don't need those backslashes.

Also put the first entry on the next line, it looks nicer.

File: code/modules/power/singularity/singularity.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/modules/power/singularity/singularity.dm:99]: Isn't spawn one of those proven to be horrible ones though

File: code/modules/power/singularity/singularity.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/modules/power/singularity/singularity.dm:125]: I've been considering standardizing object creation to implement this at tg but oldfags would whine heavily

**Comment by PJB3005** [comment]
Suggestion for the death to replace that `.dust()` call: Make the player animate being shrank into nothing. It says they're consumed from within!

**Comment by jknpj** [comment]
could use the soul stone/pylon animation. you can never get enough of that scream.

**Comment by MadmanMartian** [comment]
that's what's used currently

**Comment by MadmanMartian** [comment]
Completed PJBs recommendations, rather than delete the .dust to replace it with the shrinkdeath I've added it so it's either the dust death or the shrink death. (This is telekinetically interacting with something that consumes anything it touches, who's to know what might happen if you try to grab it with your mind)

Also reduced the chance of getting the losetta stone from 10% to 5%, as I was able to get it multiple times in one life multiple times so felt that it needed to be a smidge rarer, as it lets you potentially speak to mice or xenomorphs.

Any other ways this can be expanded upon or altered?

**Comment by Killette2** [comment]
What if instead using TK on the singulo pulled an item out of it that it sucked in? Kind of like that bug when you use TK on a chair or potted plant you can pull out an item that was inside it

**Comment by MadmanMartian** [comment]
the singularity qdels anything it 'eats', so you can't bring back anything that's been sucked in

**Comment by PJB3005** [comment]
> Kind of like that bug when you use TK on a chair or potted plant you can pull out an item that was inside it

How is that a bug?

**Comment by MadmanMartian** [comment]
he means that you can

>grab something in another room via TK
>Stuff it into a chair via TK
>Then use TK to 'search' the chair, teleporting whatever item was hidden in it into your hands regardless of distance

**Comment by PJB3005** [comment]
Oh I forgot about it teleporting, woops.

**Pull request merged**